### DISCUSSION TITLE
Aborted connections inside async message handler

### DISCUSSION THREAD

**Discussion by evanwild** [uNetworking/uWebSockets.js#1100] [Q&A]
My server is crashing when connections are closed before a send(...) call. For example:

```js
App()
  .ws('/*', {
    message: (ws, message) => {
      const response = await someFunc();
      // The socket is no longer valid
      ws.send(response);
    },

    close: (ws) => {
      // ...
    },
  });
```

How should I track which sockets have gone through the close handler? I'm using TypeScript so can't just add a new field like `ws.aborted = true`, and I'm guessing it's invalid to add some property to `getUserData()`, since the socket is invalid. Thanks.

**Comment by e3dio** [comment]
I do `ws.closed = true` don't know about Typescript did you try getUserData yet ?

**Comment by uNetworkingAB** [comment]
Have you heard about this thing called try/catch?

**Comment by uNetworkingAB** [comment]
It could also be the case that getUserData() is available at all times. Not sure.
